source/encryption.rst

```diff
@@ -0,0 +1,132 @@
+
+   To unseal Vault, execute the following command **for each Pod** of Vault
+   running:
+
+   .. code:: bash
+
+      $ kubectl exec -it pod/vault-service-0 -- vault operator unseal "$unsealKey"
+
+.. _configure-vault:
+
+Configuring Vault
+-----------------
+
+1. First, you should enable secrets within Vault. Get the Vault root token:
+
+   .. code:: bash
+
+      $ cat /tmp/vault-init | jq -r ".root_token"
+
+   The output will be like follows:
+
+   .. code:: text
+
+      s.VgQvaXl8xGFO1RUxAPbPbsfN
+
+   Now login to Vault with this token and enable the "pxc-secret" secrets path:
+
+      $ kubectl exec -it vault-0 -- /bin/sh
+      $ vault login s.VgQvaXl8xGFO1RUxAPbPbsfN
+      $ vault secrets enable --version=1 -path=pxc-secret kv
+
+   .. note:: You can also enable audit, which is not mandatory, but useful:
+
+      .. code:: bash
+
+         $ vault audit enable file file_path=/vault/vault-audit.log
+
+2. To enable Vault secret within Kubernetes, create and apply the YAML file,
+   as described further.
+
+   2.1. To access the Vault server via HTTP, follow the next YAML file example:
+
+      .. code:: text
+
+         apiVersion: v1
+         kind: Secret
+         metadata:
+           name: some-name-vault
+         type: Opaque
+         stringData:
+           keyring_vault.conf: |-
+              token = s.VgQvaXl8xGFO1RUxAPbPbsfN
+              vault_url = vault-service.vault-service.svc.cluster.local
+              secret_mount_point = pxc-secret
+
+      .. note:: the ``name`` key in the above file should be equal to the
+         ``spec.vaultSecretName`` key from the ``deploy/cr.yaml`` configuration
+         file.
+
+   2.2. To turn on TLS and access the Vault server via HTTPS, you should do two more things:
+      
+      * add one more item to the secret: the contents of the ``ca.cert`` file
+        with your certificate,
+      * store the path to this file in the ``vault_ca`` key.
+
+      .. code:: text
+
+         apiVersion: v1
+         kind: Secret
+         metadata:
+           name: some-name-vault
+         type: Opaque
+         stringData:
+           keyring_vault.conf: |-
+             token = = s.VgQvaXl8xGFO1RUxAPbPbsfN
+             vault_url = https://vault-service.vault-service.svc.cluster.local
+             secret_mount_point = pxc-secret
+             vault_ca = /etc/mysql/vault-keyring-secret/ca.cert
+           ca.cert: |-
+
+      .. note:: the ``name`` key in the above file should be equal to the
+         ``spec.vaultSecretName`` key from the ``deploy/cr.yaml`` configuration
+         file.
+         
+      .. note:: For techincal reasons the ``vault_ca`` key should either exist
+         or not exist in the YAML file; commented option like
+         ``#vault_ca = ...`` is not acceptable.
+
+More details on how to install and configure Vault can be found `in the official documentation <https://learn.hashicorp.com/vault?track=getting-started-k8s#getting-started-k8s>`_.
+
```

Comment:
please excuse me and revert https://github.com/percona/percona-xtradb-cluster-operator/pull/362/commits/1f83663189ec97dbea77000585da16e63389dea1